perl/shop/t/Basket.t

```diff
@@ -1,319 +1,364 @@
+    ok( !$hResultExists->{'Error'}, 'testExistsBasket: check error' );
+    diag $hResultExists->{'Error'}->{'Message'} . "\n" if $hResultExists->{'Error'};
 
-    return $hInfo;
+    is( $hResultExists->{'Path'},   $BasketPath, 'testExistsBasket: basket path' );
+    is( $hResultExists->{'exists'}, 1,           'testExistsBasket: exists' );
 }
 
 sub testDeleteBasket {
-    my ( $basketPath ) = @_;
+    my $BasketPath = _setupTestBasketInDB();
 
-    my $ahResults = $BasketService->delete( [$basketPath] )->result;
-    is( scalar @$ahResults, 1, 'delete: result count' );
+    my $ahResultDelete = $BasketService->delete( [$BasketPath] )->result;
+    is( scalar @$ahResultDelete, 1, 'testDeleteBasket: result count' );
 
-    my $hDelete = $ahResults->[0];
-    ok( !$hDelete->{'Error'}, 'delete: no error' );
-    diag $hDelete->{'Error'}->{'Message'}."\n" if $hDelete->{'Error'};
+    my $hResultDelete = $ahResultDelete->[0];
+    ok( !$hResultDelete->{'Error'}, 'testDeleteBasket: check error' );
+    diag $hResultDelete->{'Error'}->{'Message'} . "\n" if $hResultDelete->{'Error'};
 
-    ok( $hDelete->{'Path'} eq $basketPath, 'delete: order path' );
-    is( $hDelete->{'deleted'}, 1, 'deleted?' );
+    is( $hResultDelete->{'Path'},    $BasketPath, 'testDeleteBasket: basket path' );
+    is( $hResultDelete->{'deleted'}, 1,           'testDeleteBasket: deleted' );
 }
 
-sub testUpdateBasket {
-    my ($Path, $Basket) = @_;
-    $Basket->{Path} = $Path;
+sub testaddProductLineItem {
+    my $BasketPath = _setupTestBasketInDB();
+    my $Quantity   = 200;
 
-    my $ahResults = $BasketService->update( [$Basket] )->result;
-    is( scalar @$ahResults, 1, 'update: result count' );
+    _deleteProductsIfExists( [$Product_alias_1] );
+    _createProductsInDB(     [$Product_in_1] );
+    my $ProductGUID = _fetchProductGuid($Product_alias_1);
 
-    my $hUpdate = $ahResults->[0];
-    ok( !$hUpdate->{'Error'}, 'update: no error' );
-    diag $hUpdate->{'Error'}->{'Message'}."\n" if $hUpdate->{'Error'};
+    my $LineItem = {
+        'GUID'     => $ProductGUID,
+        'Quantity' => $Quantity,
+    };
+
+    my $ahResultAddLineItem = $BasketService->addProductLineItem( $BasketPath, [$LineItem] )->result;
+    is( scalar @$ahResultAddLineItem, 1, 'testaddProductLineItem: result count' );
 
-    ok( $hUpdate->{'Path'} eq $Path, 'path path' );
-    is( $hUpdate->{'updated'}, 1, 'updated?' );
+    my $hResultAddLineItem = $ahResultAddLineItem->[0];
+    ok( !$hResultAddLineItem->{'Error'}, 'testaddProductLineItem: check error' );
+    diag $hResultAddLineItem->{'Error'}->{'Message'} . "\n"
+      if $hResultAddLineItem->{'Error'};
 
+    is( $hResultAddLineItem->{'GUID'}, $LineItem->{GUID}, 'testaddProductLineItem: product GUID' );
+    is( $hResultAddLineItem->{'added'}, 1, 'testaddProductLineItem: added' );
 }
 
-sub testAddProductLineItem {
-    my ($Path, $LineItem, $Basket) = @_;
-    $Basket->{Path} = $Path;
+sub testGetInfoReference {
```

Comment:
Maybe we could reduce this test, if we are calling `getInfo` in the other test methods